### PR TITLE
Expand DNAME support

### PR DESCRIPTION
**Pull request by tgreenx**
## Purpose

This PR expands the functionality of DNAME resource records by giving access to its RDATA.

## Context

Necessary for https://github.com/zonemaster/zonemaster-engine/issues/568

Relates to https://github.com/zonemaster/zonemaster/issues/1075 and https://github.com/zonemaster/zonemaster/issues/472
## Changes

- New method dname() in LDNS::RR::DNAME
- Unitary test

## How to test this PR

Unit tests should pass

### CODE REVIEW

**Review by marc-vanderwal — APPROVED** [review]
Looks good to me.

File: src/LDNS.xs [hunk 1]
```diff
@@ -2459,6 +2459,18 @@ rr_cname_cname(obj)
     CLEANUP:
         free(RETVAL);
 
+MODULE = Zonemaster::LDNS        PACKAGE = Zonemaster::LDNS::RR::DNAME            PREFIX=rr_dname_
+
+char *
+rr_dname_dname(obj)
```
Comment (pnax) [inline src/LDNS.xs:2465]: suggestion: name the method `target` instead, as it seems to be the given name of the RDATA [RFC 6672 section 2](https://www.rfc-editor.org/rfc/rfc6672#section-2).

Comment (pnax) [reply src/LDNS.xs:2465]: Oh I understand. This is totally fine then.

File: lib/Zonemaster/LDNS/RR/DNAME.pm [hunk 2]
```diff
@@ -17,6 +17,10 @@ A subclass of L<Zonemaster::LDNS::RR>, so it has all the methods of that class a
 
 =head1 METHODS
 
-No RDATA methods implemented yet.
+=over
 
-=cut
+=item dname()
+
+Returns the delegation name, i.e. the <target> field from the RDATA of a DNAME record.
```
Comment (matsduf) [inline lib/Zonemaster/LDNS/RR/DNAME.pm:24]: The RDATA only consists of the target so "from the RDATA" sounds strange. There is only one field to fetch. Maybe
"Returns the delegation name, i.e. the DNAME record target (RDATA)."

**Pull request merged**